src/libraries/Microsoft.Extensions.FileSystemGlobbing/tests/FunctionalTests.cs

```diff
@@ -530,10 +518,13 @@ private void ExecuteAndVerify(Matcher matcher, string directoryPath, params stri
             var actual = results.Files.Select(match => Path.GetFullPath(Path.Combine(_context.RootPath, directoryPath, match.Path)));
             var expected = expectFiles.Select(relativePath => Path.GetFullPath(Path.Combine(_context.RootPath, relativePath)));
 
-            Assert.Equal(
-                expected.OrderBy(e => e),
-                actual.OrderBy(e => e),
-                StringComparer.OrdinalIgnoreCase);
+            AssertCollection(expected.OrderBy(e => e), actual.OrderBy(e => e));
+        }
+		
+        private static void AssertCollection(IEnumerable<string> expected, IEnumerable<string> actual)
```

Comment:
I wonder if it would make sense instead to add something to this so others could use it elsewhere?
https://github.com/dotnet/runtime/blob/c87e75e8ef260ccc6e979f9df1cc012536ee036f/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs

Eg something like `SetEquals(IEnumerable<T> expected, IEnumerable<T> actual, IComparer<T> comparer)`

I see in there there is already `public static void Equal<T>(HashSet<T> expected, HashSet<T> actual)` in therebut you'd have to make HashSets with the comparer. Plus it doesn't print a useful output (it dumps the whole lot)
AssertExtensions.Equal(new Hashset(expected), new Hashset(actual

Reply:
I don’t think you could implant out with a Hashset though as it doesn’t allow duplicates.

Reply:
Hi @danmosemsft , thanks! I fixed comment by creating CollectionEqual() in common AssertExtensions and use it in FunctionalTest.cs.
`public static void CollectionEqual<T>(IEnumerable<T> expected, IEnumerable<T> actual, IEqualityComparer<T> comparer)`
BTW, I have not updated existing HashSetEqual function, please correct me if there is misunderstanding.
`public static void Equal<T>(HashSet<T> expected, HashSet<T> actual)`

Reply:
I guess what I had in mind was that CollectionEqual would not require sorted inputs, but instead would do something simple like read both into arrays (not HashSet as they could have duplicates), assert array sizes match, then do a simple O(n^2) check that everything in the first list is found in the second. thoughts?

Reply:
@danmosemsft @maryamariyan Got the requirement now and changed code, thanks. 

There are 3 failure cases as follows:
> Xunit.Sdk.XunitException
Expected count: 3
Actual count: 2

> Xunit.Sdk.XunitException
Not found: source4.cs

> Xunit.Sdk.XunitException
Expected more: source2.cs
Actual: totally 2 source2.cs

I used hash bucket (item value -> its count) to manage actual content, so that it should be better than O(n^2) in long list case maybe, and would like to use ValueTuple but sorry I cannot find way how to refer it in this project so I created private struct ItemCount for item information: Original and Remain.

Reply:
Hi @maryamariyan , just want to ask if there is more comment so that I can continue updating? I am new at this repo and hope to learn knowledge of dotnet repo by trying to write code inside this repo. Comment welcome, thanks :)